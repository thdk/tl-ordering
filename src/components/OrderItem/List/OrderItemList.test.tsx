import { shallow, ShallowWrapper } from "enzyme";
import React from "react";
import { OrderItemList } from "..";
import ConnectedOrderItem from "./Item/OrderItem";

describe("Render OrderItemList", () => {
    let component: ShallowWrapper;
    describe("with multipe order items", () => {

        beforeEach(() => {
            component = shallow(
                <OrderItemList
                    orderId={"o1"}
                    orderItems={[
                        {
                            productId: "p1",
                            quantity: 2,
                        },
                        {
                            productId: "p2",
                            quantity: 1,
                        },
                        {
                            productId: "p3",
                            quantity: 1,
                        },
                    ]
                    }
                />,
            );
        });

        it("should match snapshot", () => {
            expect(component.debug()).toMatchSnapshot();
        });

        it("should have a OrderItemListItem component for each order item", () => {
            const items = component.find(ConnectedOrderItem);
            expect(items.length).toBe(3);
        });

        it("should display order items in correct order", () => {
            const items = component.find(ConnectedOrderItem);
            const orderIds = items.map(o => o.props().productId);

            expect(orderIds).toEqual(["p1", "p2", "p3"]);
        });
    });

    describe("without order items", () => {

        beforeEach(() => {
            component = shallow(
                <OrderItemList
                    orderId={"o1"}
                    orderItems={[]}
                />,
            );
        });

        it("should match snapshot", () => {
            expect(component.debug()).toMatchSnapshot();
        });
    });
});
