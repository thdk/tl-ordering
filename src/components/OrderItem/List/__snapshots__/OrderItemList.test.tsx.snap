// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render OrderItemList with multipe order items should match snapshot 1`] = `
"<ul>
  <Connect(OrderItem) orderId=\\"o1\\" productId=\\"p1\\" quantity={2} />
  <Connect(OrderItem) orderId=\\"o1\\" productId=\\"p2\\" quantity={1} />
  <Connect(OrderItem) orderId=\\"o1\\" productId=\\"p3\\" quantity={1} />
</ul>"
`;

exports[`Render OrderItemList without order items should match snapshot 1`] = `"<ul />"`;
